index.html

```diff
@@ -1618,52 +1649,323 @@ <h3>Methods inherited from {{ConsumedThing}}</h3>
+          </li>
+          <li>
+            Let |interaction| be the value of ||td||'s |properties|'s
+            |name|.
+          </li>
+          <li>
+            If a <a>Property</a> interaction with |name| is not found,
+            [= exception/throw =] {{NotFoundError}} and abort these steps.
+          </li>
+          <li>
+            Set the <a>internal slot</a> ||observeHandler|| on |interaction|
+            to |handler|.
           </li>
         </ol>
       </div>
     </section>
 
     <section> <h3>Handling <a>Property</a> observe requests</h3>
       <div>
-        When a network request for observing a <a>Property</a> |propertyName:string| is received by the implementation, run the following steps:
+        When a network request for observing a <a>Property</a> |name:string| is
+        received by the implementation with |options:InteractionOptions|,
+        run the following steps:
         <ol>
           <li>
-            If a <a>Property</a> with |propertyName| does not exist, return an error in the reply (as defined in the
-            <a>Thing Description</a>) and abort these steps.
+            Let |property| be the value of ||td||'s |properties|'s |name|.
+            If it does not exist, send back a {{NotFoundError}} in the
+            reply and abort these steps.
           </li>
           <li>
-            Save the request sender information to the <a>Property</a>'s <a>internal observer list</a> in order to be able to notify about <a>Property</a> value changes.
+            Save the request sender information together with |options| to
+            |property|'s <a>internal observer list</a>, in order to be able
+            to notify about <a>Property</a> value changes.
+          </li>
+          <li>
+            Every time the value of |property| changes, run the following sub-steps:
+            <ol>
+              <li>
+                Let |handler:function| be `null.`
+              </li>
+              <li>
+                If there is an ||observeHandler|| <a>internal slot</a> associated
+                with |name| on |property|, let |handler| be that.
+              </li>
+              <li>
+                Otherwise, if there is a ||readHandler|| <a>internal slot</a>
+                associated with |name| on |property|, let |handler| be that.
+              </li>
+              <li>
+                If |handler| is `null`, abort these steps.
+              </li>
+              <li>
+                Let |promise| be the result of invoking |handler| wih |options|.
+              </li>
+              <li>
+                 If |promise| rejects, abort these steps.
+              </li>
+              <li>
+                If |promise| resolves with |data|, then for each |observer| in |property|'s <a>internal observer list</a>, run the following
```

Comment:
As an exception, the steps on what to do when a Property changes are embedded in the steps to handle observe requests. I've been thinking about that, but so far I did not find a better way to split it out. They are comprised in this step: 
> Every time the value of |property| changes, run the following sub-steps:

>Why notify all the observers? 

Multiple clients (ConsumedThings) may observe a server Property. 
Since `observeProperty()` carries an `InteractionOptions`, it is saved with the subscription and applied when observe requests are handled. However, I though it does not harm if implementations also hand `option` over to the routine that generates the property change notifications (leaving open to use the `options` there as well, even though ATM `options.data` is not used in this interaction).

> How does an observer know that the interaction data returned is from a subscribe a not a normal property change?

InteractionOptions come only from `observeProperty()` calls.

> Is this difference defined in the DataSchema of an Event?

I am not sure I get the question. Property observes do not generate WoT Events. They generate protocol-specific notifications for property change, which in some protocols may involve events. We chose to not model these notifications with JS events, but with callbacks.

Reply:
Ok now is clear thank you!